### DISCUSSION TITLE
changing the default batch size

### DISCUSSION THREAD

**Discussion by ariesandrada** [mongodb/mongoid#5318] [Q&A]
I was wondering if it's possible to change the default batch size? it's mentioned in https://www.mongodb.com/docs/mongoid/master/reference/queries/#batch-size that the default is set to 1000. I looked around and can't find anything about it.

**Comment by p-mongo** [comment]
Hi @ariesandrada ,

I don't believe this is currently possible (from driver or Mongoid), can you describe your use case and what batch size you would like?

**Reply by ariesandrada** [reply]
It's just a nice to have at this point. We're currently getting cursor timeouts for some of our queries and we're manually lowering the batch size for those.

**Comment by p-mongo** [comment]
Looking at the code I don't see where the default of 1000 is defined, most likely the driver (and Mongoid) don't send any batch size by default and 1000 is actually the server's default.

**Comment by p-mongo** [comment]
I see. The recommended strategy for iterating cursors where each document takes a long time to process is to refresh the session being used, as described in https://jira.mongodb.org/browse/DOCS-11255.

**Reply by ariesandrada** [reply]
Aren't cursor and session timeouts 2 different things? And how can this be done in Ruby?

**Reply by p-mongo** [reply]
Correct. The issue is that even though a cursor can be created without a timeout, the session timeout still applies to the cursor using that session, and as of 3.6 server the driver uses sessions implicitly for all operations.

We don't presently have the Ruby driver documentation equivalent of that snippet but I filed https://jira.mongodb.org/browse/RUBY-3028 to translate it into Ruby and add to our documentation.
